File: 1503/1503.go
func lastMomentBeforeAllAntsFallOut(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := lastMomentBeforeAllAntsFallOut(root.Left)
	r := lastMomentBeforeAllAntsFallOut(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 1503/1503.cs
public class Solution {
    public int LastMomentBeforeAllAntsFallOut(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(LastMomentBeforeAllAntsFallOut(root.left), LastMomentBeforeAllAntsFallOut(root.right));
    }
}